tee-worker/omni-executor/rpc-server/src/methods/omni/get_hyperliquid_signature_data.rs

```diff
@@ -0,0 +1,508 @@
+				})?;
+
+			let identity = Identity::try_from(params.user_id.clone()).map_err(|e| {
+				error!("Failed to convert user ID to identity: {}", e);
+				ErrorObject::from(ErrorCode::ParseError)
+			})?;
+			let omni_account = identity.to_omni_account(&params.client_id);
+
+			let derived_pubkey = ctx
+				.signer_client
+				.request_wallet(ChainType::Evm, 0, *omni_account.as_ref())
+				.await
+				.map_err(|_| {
+					error!("Failed to derive EVM address");
+					ErrorObject::from(ErrorCode::InternalError)
+				})?;
+			let main_address =
+				pubkey_to_address(ChainType::Evm, &derived_pubkey).map_err(|_| {
+					error!("Failed to convert derived pubkey to address");
+					ErrorObject::from(ErrorCode::InternalError)
+				})?;
+
+			let nonce = Utc::now().timestamp_millis() as u64;
+
+			let hyperliquid_chain = if is_testnet_chain(params.chain_id) {
+				"Testnet".to_string()
+			} else {
+				"Mainnet".to_string()
+			};
+
+			let (action, signature) = match params.action_type {
+				HyperliquidActionType::ApproveAgent { agent_address, agent_name } => {
+					let action = ApproveAgentAction {
+						signature_chain_id: params.chain_id,
+						hyperliquid_chain,
+						agent_address: Address::from_str(&agent_address).map_err(|_| {
+							error!("Invalid agent address format");
+							ErrorObject::from(ErrorCode::ParseError)
+						})?,
+						agent_name,
+						nonce,
+					};
+					let signature =
+						generate_eip712_signature(&ctx, &action, omni_account.as_ref()).await?;
+					(HyperliquidAction::ApproveAgent(action), signature)
+				},
+				HyperliquidActionType::Withdraw3 { amount, destination } => {
+					let action = Withdraw3Action {
+						signature_chain_id: params.chain_id,
+						hyperliquid_chain,
+						amount,
+						time: nonce,
+						destination: Address::from_str(&destination).map_err(|_| {
+							error!("Invalid destination address format");
+							ErrorObject::from(ErrorCode::ParseError)
+						})?,
+					};
+					let signature =
+						generate_eip712_signature(&ctx, &action, omni_account.as_ref()).await?;
+					(HyperliquidAction::Withdraw3(action), signature)
+				},
+				HyperliquidActionType::ApproveBuilderFee { max_fee_rate, builder } => {
+					let action = ApproveBuilderFeeAction {
+						signature_chain_id: params.chain_id,
+						hyperliquid_chain,
+						max_fee_rate,
+						builder: Address::from_str(&builder).map_err(|_| {
+							error!("Invalid builder address format");
+							ErrorObject::from(ErrorCode::ParseError)
+						})?,
+						nonce,
+					};
+					let signature =
+						generate_eip712_signature(&ctx, &action, omni_account.as_ref()).await?;
+					(HyperliquidAction::ApproveBuilderFee(action), signature)
+				},
+			};
+
+			Ok(GetHyperliquidSignatureDataResponse {
+				main_address,
+				hyperliquid_signature_data: HyperliquidSignatureData { action, nonce, signature },
+			})
+		})
+		.expect("Failed to register omni_getHyperliquidSignatureData method");
```

Comment:
Better error handling?

Reply:
I think this consistent with other RPC calls.

Reply:
I see -then probably needs to be solved together with other methods